src/Marten.Testing/Bugs/Bug_PR_1412_spaces_in_table_name_cause_schema_patch_to_fail.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Marten.Schema;
+using Marten.Storage;
+using Xunit;
+
+namespace Marten.Testing.Bugs
+{
+    public class Bug_PR_1412_spaces_in_table_name_cause_schema_patch_to_fail : IntegratedFixture
+    {
+
+        internal class testSchema : IFeatureSchema
+        {
+            public ISchemaObject[] Objects => SchemaObjects().ToArray();
+
+            public Type StorageType => GetType();
+
+            public string Identifier { get; set; }
+
+            public IEnumerable<Type> DependentTypes()
+            {
+                return new List<Type>();
+            }
+
+            public bool IsActive(StoreOptions options)
+            {
+                return true;
+            }
+
+            public void WritePermissions(DdlRules rules, StringWriter writer)
+            {
+                // No permissions!
+            }
+
+            protected virtual IEnumerable<ISchemaObject> SchemaObjects()
+            {
+                List<ISchemaObject> objects = new List<ISchemaObject>();
+                return objects;
+            }
+
+        }
+
+        internal class spaceAfterTableNameSchema: testSchema
+        {
+            protected override IEnumerable<ISchemaObject> SchemaObjects()
+            {
+                var table = new Table(new DbObjectName("test_space_after"));
+                table.AddColumn("space_after ", "int");
+                return new List<ISchemaObject>
+                {
+                        table
+                };
+            }
+        }
+        [Fact]
+        public void space_after_table_name_does_not_cause_exception()
+        {
+            StoreOptions(_ =>
+            {
+                _.AutoCreateSchemaObjects = AutoCreate.CreateOrUpdate;
+                _.Storage.Add(new spaceAfterTableNameSchema());
+            });
+            try
+            {
+                theStore.Schema.ApplyAllConfiguredChangesToDatabase();
+                theStore.Schema.ApplyAllConfiguredChangesToDatabase();
```

Comment:
**CONCERN:** @malscent why is it needed to apply changes to database twice? Same concern applies for the tests below.

Reply:
Because the error here is that the first application of the changes is s uccessful and Postgres truncates the the whitespace.  The second apply is the one that throws, due to Marten not being able to generate the schema diff.

Reply:
Thank you for an explanation. Maybe it would be worth to add a comment with information or update the test name to mention this second call?

Reply:
I was also wondering about the a call twice, thanks for the clarification. A comment will add clarity.

Reply:
@malscent I think the second call can be changed to `theStore.Schema.AssertDatabaseMatchesConfiguration()`. This results in the same check which is equivalent and generates the schema diff. This way an explicit comment/explanation is not required. Thoughts?

Reply:
> I added a comment to use `AssertDatabaseMatchesConfiguration()` for the second call, rest all looks fine.

Yup.. Adding now